pkg/cmd/admin/migrate/policy/policy.go

```diff
@@ -0,0 +1,255 @@
+			ErrOut:        errout,
+			AllNamespaces: true,
+			Include:       []string{"clusterrole", "role", "clusterrolebinding", "rolebinding"},
+		},
+	}
+	cmd := &cobra.Command{
+		Use:   name,
+		Short: "Confirm that Origin policy resources are in sync with their RBAC equivalents",
+		Long:  internalMigratePolicyLong,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(name, f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			kcmdutil.CheckErr(options.Run())
+		},
+	}
+	return cmd
+}
+
+func (o *MigratePolicyOptions) Complete(name string, f *clientcmd.Factory, c *cobra.Command, args []string) error {
+	if len(args) != 0 {
+		return fmt.Errorf("%s takes no positional arguments", name)
+	}
+
+	if err := o.ResourceOptions.Complete(f, c); err != nil {
+		return err
+	}
+
+	_, kclient, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	o.rbac = kclient.Rbac()
+
+	return nil
+}
+
+func (o MigratePolicyOptions) Validate() error {
+	return o.ResourceOptions.Validate()
+}
+
+func (o MigratePolicyOptions) Run() error {
+	return o.ResourceOptions.Visitor().Visit(func(info *resource.Info) (migrate.Reporter, error) {
+		return o.checkParity(info.Object)
+	})
+}
+
+// checkParity confirms that Openshift policy objects are in sync with Kubernetes RBAC
+// and returns an error if they are out of sync or if it encounters a conversion error
+func (o *MigratePolicyOptions) checkParity(obj runtime.Object) (migrate.Reporter, error) {
+	var errlist []error
+	switch t := obj.(type) {
+	case *authorizationapi.ClusterRole:
+		errlist = append(errlist, o.checkClusterRole(t)...)
+	case *authorizationapi.Role:
+		errlist = append(errlist, o.checkRole(t)...)
+	case *authorizationapi.ClusterRoleBinding:
+		errlist = append(errlist, o.checkClusterRoleBinding(t)...)
+	case *authorizationapi.RoleBinding:
+		errlist = append(errlist, o.checkRoleBinding(t)...)
+	default:
+		return nil, nil // indicate that we ignored the object
+	}
+	return notChanged, utilerrors.NewAggregate(errlist) // we only perform read operations
+}
+
+func (o *MigratePolicyOptions) checkClusterRole(originClusterRole *authorizationapi.ClusterRole) []error {
+	var errlist []error
+
+	// convert the origin role to a rbac role
+	convertedClusterRole := &rbac.ClusterRole{}
+	if err := authorizationapi.Convert_api_ClusterRole_To_rbac_ClusterRole(originClusterRole, convertedClusterRole, nil); err != nil {
+		errlist = append(errlist, err)
+	}
+
+	// try to get the equivalent rbac role from the api
+	rbacClusterRole, err := o.rbac.ClusterRoles().Get(originClusterRole.Name, v1.GetOptions{})
+	if err != nil {
+		errlist = append(errlist, err)
+	}
+
+	// compare the results if there have been no errors so far
+	if len(errlist) == 0 {
+		// there's one wrinkle.  If `openshift.io/reconcile-protect` is to true, then we must set rbac.authorization.kubernetes.io/autoupdate to false
+		if convertedClusterRole.Annotations["openshift.io/reconcile-protect"] == "true" {
```

Comment:
this looks very familiar.  Restructure enough to be able to re-use the implementation for this with the controllers.